wandb/data_types.py

```diff
@@ -871,18 +895,202 @@ def seq_to_json(cls, images, run, key, step):
+    """
+    Wandb class for 2D bounding Boxes
+    """
+
+    def type_name(self):
+        return "boxes2D"
+
+    def validate(self , boxes):
+        if not isinstance(boxes, collections.Sequence):
+            raise TypeError("Boxes must be a list")
+
+        for box in boxes:
+            # Required arguments
+            error_str = "Each box must contain a position with: middle, width, and height or \
+                    \nminX, maxX, minY, maxY."
+            if not "position" in box:
+                raise TypeError(error_str)
+            else:
+                # import ipdb; ipdb.set_trace()
+                valid = False
+                if "middle" in box["position"] and len(box["position"]["middle"]) == 2 and \
+                   has_num(box["position"], "width") and \
+                   has_num(box["position"], "height"): 
+                   valid = True
+                elif has_num(box["position"], "minX") and \
+                     has_num(box["position"], "maxX") and \
+                     has_num(box["position"], "minY") and \
+                     has_num(box["position"], "maxY"):
+                   valid = True
+                
+                if not valid:
+                    raise TypeError(error_str)
+
+
+            # Optional arguments
+            if ("scores" in box) and not isinstance(box["scores"], dict):
+                raise TypeError("each")
+            else:
+                for k,v in list(box["scores"].items()):
+                    if not isinstance(k, six.string_types):
+                        raise TypeError("A score key must be a string")
+                    if not isinstance(v, numbers.Number):
+                        raise TypeError("A score value must be a number")
+
+            if ("class_label" in box) and not isinstance(box["class_label"], six.string_types):
+                raise TypeError("A box's class label must be of type must be of type string")
+
+            # Optional
+            if ("box_caption" in box) and not isinstance(box["box_caption"], six.string_types):
+                raise TypeError("A box's caption must be a string")
+
+class ImageMask(Media):
+    """
+    Wandb class for image masks, useful for segmentation tasks
+    """
+
+    def __init__(self, val, **kwargs):
+        super(ImageMask, self).__init__()
+
+        self.validate(val)
+        self._val = val
+
+        ext = "." + self.type_name() + ".png"
+        tmp_path = os.path.join(MEDIA_TMP.name, util.generate_id() + ext)
+
+        PILImage = util.get_module(
+            "PIL.Image", required='wandb.Image needs the PIL package. To get it, run "pip install pillow".')
+        image = PILImage.fromarray(Image.to_uint8(val["mask_data"]), mode="L")
+        
+        image.save(tmp_path, transparency=None)
+        self._set_file(tmp_path, is_tmp=True, extension=ext)
+
+    def get_media_subdir(self):
+        return os.path.join('media', 'metadata', self.type_name())
```

Comment:
Since we have type_name in the image name already do we really need a sub dir?  What about `media/images/masks` or just `media/masks`?